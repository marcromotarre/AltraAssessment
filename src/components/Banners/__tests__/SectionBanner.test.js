import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import SectionBanner from '../SectionBanner';

/** this component should:
 *  - render Main Banner with:
 *      - town name
 *      - filter icon
 *      - svg background
 */

const props = {
  townName: 'town name',
};

describe('Section Banner Component tests', () => {
  it('SectionBanner Snapshot', () => {
    const tree = renderer.create(<SectionBanner />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('SectionBanner renders Link and HomeIcon', () => {
    const wrapper = shallow(<SectionBanner />);
    expect(wrapper.children().length).toBe(1);
    const link = wrapper.childAt(0);
    expect(link.type().displayName).toBe('Link');
    const homeIcon = link.childAt(0);
    expect(homeIcon.type()).toBe('svg');
    expect(homeIcon.prop('width')).toBe(25);
  });
});
