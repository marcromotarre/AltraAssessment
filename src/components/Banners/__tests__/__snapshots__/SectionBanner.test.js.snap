// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Banner Component tests SectionBanner Snapshot 1`] = `
<div
  className="sc-AxjAm hCXQxe"
>
  <a
    aria-current="page"
    href="/"
    onClick={[Function]}
  >
    <svg
      width={25}
    />
  </a>
</div>
`;
